### DISCUSSION TITLE
Reload Nginx from outside of the container?

### DISCUSSION THREAD

**Discussion by leoborlot** [nginx-proxy/nginx-proxy#1806] [Q&A] [answered]
hi.

We are building a new infra here, and we find a problem, some times we need to modify a container but this change don´t was alerting the docker socket and cannot was updating the nginx. 
When we restart the nginx container manually, it become up update and the modifiiation was found. 
There any way to restart or execute a reload from Xginx service to it callback about the container list and restart the operation with new information abount our application container?

I´ve try some another poublications but, its is many old, and say about docker-gen. I´ve try this withou sucess. 

Any ont have any soluction?

**Comment by Dr-Shadow** [comment, marked as answer]
For anyone having issues with reload, you can signal the process you need with these commands :

```
docker exec -it nginx-proxy sh -c 'kill -SIGHUP $(pidof docker-gen)'
docker exec -it nginx-proxy sh -c 'kill -SIGHUP $(cat /run/nginx.pid)'
```

Of course, you will probably need to escape the $ or ' depending on your use case (in scripts, etc...)
